cypress/e2e/common/sitemapSupport.js

```diff
@@ -0,0 +1,110 @@
+/// <reference types="Cypress" />
+import { Given, Then } from "cypress-cucumber-preprocessor/steps";
+
+Given("user requests the sitemap from {string}", (url) => {
+    cy.request(url).as('sitemap_response');
+});
+
+Then("the response content type is {string}", (expectedType) => {
+
+    cy.get('@sitemap_response').should( response =>  {
+
+        expect(response).to.have.property('headers');
+        const actualType = response.headers['content-type'];
+
+        expect(actualType).to.equal(expectedType);
+    });
+});
+
+
+Then("the sitemap has at least {int} urls", (minimumCount) => {
```

Comment:
I would suggest to combine the min and max step into one - seeing how the code is duplicated, and only assertion portion is different 

Then('the sitemap has at least {int} and no more than {int} urls',(minimumCount, maximumCount)=>{

    cy.get("@sitemap_response").should((response) => {
        expect(response).to.have.property('body');

        const urls = Cypress.$(response.body)
            .find("loc")
            .toArray()
            .map((el) => el.innerText);

        expect(urls.length).to.be.at.least(minimumCount);
        expect(urls.length).to.be.lessThan(maximumCount)
    });
});

Reply:
That's fair.